compiler/prelude/prelude.go

```diff
@@ -372,7 +372,7 @@ var $internalAppend = function(slice, array, offset, length) {
 
 var $equal = function(a, b, type) {
   if (type === $jsObjectPtr) {
-    return a === b;
+    return a === b || (a.hasOwnProperty("object") && a.object === undefined && b.hasOwnProperty("object") && b.object === undefined);
```

Comment:
As I've written in the other comments, the `*js.Object` type holds the JS value itself. So when comparing two `undefined`, the JS value in `a` would be `undefined` and `b` would also be `undefined`. So `a === b` holds.